crates/cxx-qt-lib/src/core/qpoint.rs

```diff
@@ -45,7 +46,7 @@ mod ffi {
 }
 
 /// The QPoint struct defines a point in the plane using integer precision.
-#[derive(Debug, Clone, PartialEq, Eq)]
+#[derive(Debug, Clone, PartialEq, Eq, Add, Sub, Mul)]
 #[repr(C)]
 pub struct QPoint {
```

Comment:
Hmm i was assuming we'd use the C++ operators.

Eg QPoint has += with a QPoint or *= with a factor. Then QRect has += with QMargins etc :thinking: 

And we have the debate in other PRs about if we should rely on the fields or the Qt implementation. Eg for QPoint elsewhere instead of making the field `x` public we have made the method `x()` as we aren't sure if the Qt inner implementation could ever change :thinking:

Reply:
CXX is also planning to add a "derive" for C++ operators in future:
https://github.com/dtolnay/cxx/issues/109

Maybe something we should upstream :thinking: 

Unfortunately at the moment something like this doesn't work:
``` rust
    unsafe extern "C++" {
        #[cxx_name = "operator+="]
        fn add(self: &mut QSize, other: &QSize) -> &mut QSize;
    }
```
CXX prevents you from using `operator+=` as the name of the function.

Reply:
Right for now creating a template in C++ like we have for constructor/drop etc would be the workaround.

Reply:
> as we aren't sure if the Qt inner implementation could ever change

Ah I see, I will change it to use an template on the C++ side.

Reply:
In the serde one i'm also have a similar debate for simple types like QPoint, should we just use derive and the x, y. Or impl (de)seralise and read from the methods x() and y().

Not relying on the Qt structure but public API is probably more robust and means things don't need to be rewritten when Qt versions change stuff ...